File: dcp-solutions/2024-09-29/solution.c
```c
// Definition for singly-linked list.
typedef struct ListNode {
    int val;
    struct ListNode *next;
} ListNode;

ListNode* getIntersectionNode(ListNode* headA, ListNode* headB) {
    // Calculate the lengths of both lists
    int lenA = 0;
    ListNode* tempA = headA;
    while (tempA != NULL) {
        lenA++;
        tempA = tempA->next;
    }

    int lenB = 0;
    ListNode* tempB = headB;
    while (tempB != NULL) {
        lenB++;
        tempB = tempB->next;
    }

    // Move the longer list forward
    if (lenA > lenB) {
        for (int i = 0; i < lenB; i++) {
            headA = headA->next;
        }
    } else {
        for (int i = 0; i < lenA; i++) {
            headB = headB->next;
        }
    }

    // Move both lists forward until they meet
    while (headA != NULL && headB != NULL) {
        if (headA == headB) return headA;
        headA = headA->next;
        headB = headB->next;
    }

    // If no intersection is found, return NULL
    return NULL;
}
```

File: dcp-solutions/2024-09-29/solution.py
```python
# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

def get_intersection_node(headA, headB):
    # Calculate the length of linked list A
    len_A = 0
    temp_A = headA
    while temp_A is not None:
        len_A += 1
        temp_A = temp_A.next

    # Move pointer A to the end of its list and move pointer B by the difference in lengths
    for i in range(len_A):
        if headB is not None:
            headB = headB.next

    # At this point, both pointers are at the start of their lists. 
    # If they meet again, it will be at the intersection point.
    while headA is not None and headB is not None:
        if headA == headB:
            return headA
        headA = headA.next
        headB = headB.next

    # If no intersection point found
    return None
```